ui/src/app/views/notebook-resource-card.tsx

```diff
@@ -0,0 +1,188 @@
+import {workspacesApi} from 'app/services/swagger-fetch-clients';
+import colors from 'app/styles/colors';
+import {navigateByUrl} from 'app/utils/navigation';
+import {ResourceType} from 'app/utils/resourceActions';
+import {CopyNotebookModal} from 'app/views/copy-notebook-modal';
+import {RenameModal} from 'app/views/rename-modal';
+import {Action, ResourceCardTemplate} from 'app/views/resource-card-template';
+import {RecentResource} from 'generated/fetch';
+import * as React from 'react';
+import * as fp from 'lodash';
+import {withErrorModal} from "app/views/with-error-modal";
+import {withConfirmDeleteModal} from "app/views/with-confirm-delete-modal";
+
+interface Props {
+  resourceCard: RecentResource; // Destructure this into used parameters only
+  onUpdate: Function; // TODO eric: this could use a better name
```

Comment:
Obviously super-early nitpick, but these functions should eventually have tighter type-definitions. I've been seeing a fair amount of "Function" type defs elsewhere in our codebase, which leave us open to all sorts of confusion and type-related bugs :)

Reply:
sounds good. I actually didnt know you could specify function types. I'll swap it out wherever I can